src/convert/convertContext.ts

```diff
@@ -193,12 +194,21 @@ class NonDecompositionFinalizer extends ConvertTransactionFinalizer<NonDecomposi
       return writerData;
     }
     this.tree = tree;
+
+    const packageDirectories = SfProject.getInstance().getPackageDirectories();
+    const pkgPaths = packageDirectories.map((pkg) => pkg.fullPath);
+
     // nondecomposed metadata types can exist in multiple locations under the same name
     // so we have to find all components that could potentially match inbound components
-    const allNonDecomposed = this.getAllComponentsOfType(
-      defaultDirectory,
-      this.transactionState.exampleComponent.type.name
-    );
+    let allNonDecomposed: SourceComponent[];
+
+    if (pkgPaths.includes(defaultDirectory)) {
+      allNonDecomposed = this.getAllComponentsOfType(pkgPaths, this.transactionState.exampleComponent.type.name);
+    } else {
+      // defaultDirectory isn't a package, assumes it's the target output dir for conversion
+      // so no need to scan this folder
+      allNonDecomposed = [];
```

Comment:
This is the output dir passed to the converter in the conversion pipeline, if the dir isn't a package in a project SDR don't need to scan it.

https://github.com/forcedotcom/source-deploy-retrieve/blob/3af34cedbccb2725e934eceda82331798030ba82/src/convert/metadataConverter.ts#L116

When running `sfdx force:mdapi:convert -r mdapi -d src` SDR will not look for non-decomposed metadata in `/home/user/dreamhouse-lwc/src`.